sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,261 @@
+ *    &lt;property name="methodName" value="assert(True|False)"/&gt;
+ *    &lt;property name="argumentCount" value="1"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="ForbidCertainMethodsCheck"&gt;
+ *    &lt;property name="methodName" value="assertEquals"/&gt;
+ *    &lt;property name="argumentCount" value="2"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * Argument count can be bounded range (e.g.: {@code 2-4}) or unbounded range
+ * (e.g.: {@code -5, 6-}). Unbounded range can be unbounded only on one side.
+ * Multiple ranges must be comma separated.
+ * For example, the following will allow only 4 and 8 arguments.
+ *
+ * <pre>
+ * &lt;module name="ForbidCertainMethodsCheck"&gt;
+ *    &lt;property name="methodName" value="asList"/&gt;
+ *    &lt;property name="argumentCount" value="-3, 5-7, 9-"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>Note: Because of limitations of how checkstyle works we can't check type of the
+ * arguments.
+ * </p>
+ * @author <a href="mailto:raghavgautam@gmail.com">Raghav Kumar Gautam</a>
+ */
+public class ForbidCertainMethodsCheck extends AbstractCheck {
+
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_WITH_ARG = "forbid.certain.methods";
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY_WITHOUT_ARG = "forbid.certain.methods.noarg";
+    /** Regex for splitting string on comma. */
+    private static final Pattern COMMA_REGEX = Pattern.compile(",");
+
+    /** Name of the method. */
+    private Pattern methodName;
+
+    /** Range for number of arguments. */
+    private String argumentCount;
+    /** Range objects for matching number of arguments. */
+    private final List<IntRange> argumentCountRanges = new ArrayList<>();
+
+    /**
+     * Set method name regex for the forbidden method.
+     * @param methodName regex for the method name
+     * @throws CheckstyleException when methodName is not a valid regex
+     */
+    public void setMethodName(String methodName) throws CheckstyleException {
+        this.methodName = CommonUtils.createPattern(methodName);
+    }
+
+    /**
+     * Set range to match number of arguments of the forbidden method. It can be null.
+     * @param argumentCount range for matching number of arguments
+     * @throws CheckstyleException when argumentCount is not a valid range
+     */
+    public void setArgumentCount(String argumentCount) throws CheckstyleException {
+        this.argumentCount = argumentCount;
+        if (argumentCount.trim().length() > 0) {
+            final String[] rangeTokens = COMMA_REGEX.split(argumentCount);
+            for (String oneToken : rangeTokens) {
+                this.argumentCountRanges.add(new IntRange(oneToken));
+            }
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
```

Comment:
Please move this definition to getRequired method and getRequired in other methods

Reply:
Done.